src/components/chart/chartHelpers.js

```diff
@@ -0,0 +1,403 @@
+ * @param {number} params.xAxisLabelIncrement
+ * @param {object} params.xAxisPropDefaults
+ * @param {Function} params.xAxisTickFormat
+ * @returns {{tickFormat: (function(*)), tickValues: *}}
+ */
+const generateXAxisProps = ({
+  dataSet = {},
+  maxX,
+  xAxisLabelIncrement,
+  xAxisPropDefaults = {},
+  xAxisTickFormat
+} = {}) => {
+  const { data = [] } = dataSet;
+  const axisProps = {
+    ...xAxisPropDefaults,
+    tickValues: data.reduce(
+      (acc, current, index) => (index % xAxisLabelIncrement === 0 ? acc.concat(current.x) : acc),
+      []
+    ),
+    tickFormat: tick => data[tick]?.xAxisLabel || tick
+  };
+
+  if (typeof xAxisTickFormat === 'function') {
+    axisProps.tickFormat = tick => {
+      const tickIndex = axisProps.tickValues.indexOf(tick);
+      const previousItem = { ...data[axisProps.tickValues[tickIndex - 1]] };
+      const nextItem = { ...data[axisProps.tickValues[tickIndex + 1]] };
+      const item = { ...data[tick] };
+
+      return xAxisTickFormat({ tick, previousItem, item, nextItem, maxX });
+    };
+  }
+
+  return axisProps;
+};
+
+/**
+ * Generate Y axis props, ticks, tick formatting.
+ *
+ * @param {object} params
+ * @param {Array} params.dataSets
+ * @param {number|object} params.maxY
+ * @param {object} params.yAxisPropDefaults
+ * @param {Function} params.yAxisTickFormat
+ * @returns {Array}
+ */
+const generateYAxisProps = ({ dataSets = [], maxY, yAxisPropDefaults = {}, yAxisTickFormat } = {}) => {
+  const axisProps = [];
+  const isMultiAxis = dataSets.length > 1;
+
+  dataSets.forEach(({ id, stroke, strokeWidth } = {}, index) => {
+    const updatedAxisProps = {
+      style: { axis: {}, tickLabels: {} },
+      tickFormat: tick => tick
+    };
+
+    if (isMultiAxis && stroke) {
+      updatedAxisProps.style.axis.stroke = stroke;
+    }
+
+    if (isMultiAxis && strokeWidth) {
+      updatedAxisProps.style.axis.strokeWidth = strokeWidth;
+    }
+
+    if (typeof yAxisTickFormat === 'function') {
+      const updatedMaxY = (typeof maxY === 'number' && maxY) || maxY?.[id];
+
+      updatedAxisProps.tickFormat = tick => {
+        const normalizedTick = (isMultiAxis && tick * updatedMaxY) || tick;
+
+        return yAxisTickFormat({
+          tick: normalizedTick,
+          isMultiAxis,
+          maxY: updatedMaxY
+        });
+      };
+    }
+
+    axisProps.push({
+      ...yAxisPropDefaults,
+      ...updatedAxisProps,
+      orientation: (index === 0 && 'left') || 'right'
+    });
+  });
+
+  return axisProps;
+};
+
+/**
+ * Generate x,y props.
+ *
+ * @param {object} params
+ * @param {Array} params.dataSets
+ * @param {object} params.individualMaxY
+ * @param {number} params.maxX
+ * @param {number} params.maxY
+ * @param {boolean} params.xAxisFixLabelOverlap
+ * @param {number} params.xAxisLabelIncrement
+ * @param {Function} params.xAxisTickFormat
+ * @param {Function} params.yAxisTickFormat
+ * @returns {{xAxisProps: object, yAxisProps: Array}}
+ */
+const generateAxisProps = ({
```

Comment:
optional: plural of Axis is Axes... this wording also works in the singular spelling